client/pages/trip-travelers.jsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import * as Icons from '../components/svg';
+import { TopNav, Footer } from '../components/navigation';
+import AddTraveler from '../components/form-addtraveler';
+
+export default class TripTravelers extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentTrip: null,
+      modalView: null,
+      travelers: []
+    };
+    this.setState = this.setState.bind(this);
+    this.handleClickAddTraveler = this.handleClickAddTraveler.bind(this);
+  }
+
+  componentDidMount() {
+    this.getSingleTrip();
+    this.getTravelersList();
+  }
+
+  getSingleTrip() {
+    fetch(`/api/trip/${this.props.tripId}`)
+      .then(response => response.json())
+      .then(trips => {
+        this.setState({ currentTrip: trips });
+      })
+      .catch(err => console.error(err));
+  }
+
+  getTravelersList() {
+    fetch(`/api/travelers/${this.props.tripId}`)
+      .then(response => response.json())
+      .then(travelersList => {
+        this.setState({ travelers: travelersList });
+      })
+      .catch(err => console.error(err));
+  }
+
+  handleClickAddTraveler() {
+    this.setState({
+      modalView: true
+    });
+  }
+
+  render() {
+    if (!this.state.currentTrip) return null;
+    const { name } = this.state.currentTrip;
+    const { modalView, travelers } = this.state;
+    const renderModal = () => {
+      if ((modalView === null && travelers.length === 0) || modalView) {
+        return <AddTraveler view={modalView} tripId={this.props.tripId} travelers={travelers} TravelerSetState={this.setState} />;
```

Comment:
In general we do not want do pass `setState` down into other components. Instead you should make the post request inside of `TripTravelers`. Give `TripTravelers` a method that accepts a traveler object and does the post request. `AddTraveler` should really only be responsible for handling the user input and then sending that user input back up to `TripTravelers`.